src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/KeyMomentEvaluator.cs

```diff
@@ -0,0 +1,65 @@
+﻿
+using System.Collections.Generic;
+using JetBrains.Annotations;
+using RegressionGames.StateRecorder.BotSegments.Models;
+using RegressionGames.StateRecorder.KeyMoments;
+using RegressionGames.StateRecorder.Models;
+
+namespace RegressionGames.StateRecorder
+{
+    public class KeyMomentEvaluator
+    {
+
+        private long _keyMomentNumber;
+
+        public void Reset()
+        {
+            _keyMomentNumber = 1;
+        }
+
+        // Future: We may change this to allow easier registration of evaluators by scanning the classpath... we already do a pattern like this in other code in the SDK, so should be easy to add in later
+        // Most of this class is where future work for new recording of key moments for keyboard inputs or actions will be glued in.  This class WILL change its implementation quite a bit as those pieces are added and we see what patterns work best.
+        private readonly IKeyMomentEvaluator[] _keyMomentEvaluators = {
+            new MouseKeyMomentEvaluator()
+        };
+
+        [CanBeNull]
+        public BotSegment EvaluateKeyMoment(long tickNumber, out long keyMomentNumber)
```

Comment:
This is where `KeyMoment` recording happens.

We will hook in other types beyond mouse as noted in the comments in this file.